trl/trainer/reward_trainer.py

```diff
@@ -98,15 +98,26 @@ def __init__(
             peft_config (`Dict`, defaults to `None`):
                 The PEFT configuration to use for training. If you pass a PEFT configuration, the model will be wrapped in a PEFT model.
         """
-        if max_length is not None and args.max_length is not None:
-            raise ValueError(
-                "You cannot specify both `max_length` and `args.max_length`. Please use the `RewardConfig` to set `max_length` once."
-            )
-        if max_length is not None and args.max_length is None:
+        if type(args) == TrainingArguments:
```

Comment:
```suggestion
        if isinstance(args, TrainingArguments):
```

Reply:
I tried this initially, but realised it wont' work because `args` is a subclass of `TrainingArguments` and thus is an instance of `TrainingArguments` which makes the if-statement always true.  See e.g. this: https://stackoverflow.com/questions/1549801/what-are-the-differences-between-type-and-isinstance

Happy to refactor to a try / except clause if you prefer :)

Reply:
I see perfect, thanks a lot lewis for explaining!